File: app/src/main/java/org/tat/sdksample/recommendedroute/search/RecommendedRouteActivity.java
package org.tat.sdksample.recommendedroute.search;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.ActionBar;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;
import android.view.View;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.Spinner;
import android.widget.Switch;

import org.tat.sdksample.R;
import org.tat.sdksample.recommendedroute.list.RecommendedRouteListActivity;
import org.tat.sdksample.recommendedroute.model.RecRoutes;
import org.th.tatsdk.ServiceRequestListener;
import org.th.tatsdk.common.TATLanguage;
import org.th.tatsdk.common.TATPoint;
import org.th.tatsdk.common.TATRegion;
import org.th.tatsdk.route.list.TATGetRoutes;
import org.th.tatsdk.route.list.TATGetRoutesParameter;
import org.th.tatsdk.route.list.TATGetRoutesResultSet;

public class RecommendedRouteActivity extends AppCompatActivity {

    Spinner spinnerDay;
    Spinner spinnerRegion;
    Switch swNearby;
    Button btnSearch;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_recommended_route);
        initInstances();
        setupToolbar();
    }

    private void setupToolbar() {
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            actionBar.setTitle(R.string.recommended_routes);
            actionBar.setHomeButtonEnabled(true);
            actionBar.setDisplayHomeAsUpEnabled(true);
        }
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                break;
            default:
                break;
        }
        return true;
    }

    private void initInstances() {
        spinnerDay = findViewById(R.id.spinnerDay);
        spinnerRegion = findViewById(R.id.spinnerRegion);
        swNearby = findViewById(R.id.swNearbyRecRoute);
        btnSearch = findViewById(R.id.btn_search_rectoute);
        btnSearch.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                loadRecRoute();
            }
        });
        String[] days = getResources().getStringArray(R.array.day);
        spinnerDay.setAdapter(new ArrayAdapter<>(getBaseContext(), android.R.layout.simple_list_item_1, days));
        String[] region = getResources().getStringArray(R.array.region);
        spinnerRegion.setAdapter(new ArrayAdapter<>(getBaseContext(), android.R.layout.simple_list_item_1, region));

    }

    private void loadRecRoute() {
        TATGetRoutesParameter param = new TATGetRoutesParameter(TATLanguage.ENGLISH);
        String region = spinnerRegion.getSelectedItem().toString();
        int day = spinnerDay.getSelectedItemPosition();
        if (!region.equals("All")) {
            TATRegion selectedRegion = stringToRegion(region);
            if (selectedRegion != null) {
                param.setRegion(selectedRegion);
            }
        }

        //set number of trip day
        if (day != 0) {
            param.setNumberOfDays(day);
        }

        //set search center location
        if (swNearby.isChecked()) {
            param.setLatitude(13.74918);
            param.setLongitude(100.55785);
        }

        TATGetRoutes.executeAsync(param, new ServiceRequestListener<TATGetRoutesResultSet>() {

            @Override
            public void onResponse(TATGetRoutesResultSet result) {
                RecRoutes res = new RecRoutes(result.getResults());
                Intent intent = new Intent(getBaseContext(), RecommendedRouteListActivity.class);
                intent.putExtra("route_list", res);
                startActivity(intent);
            }

            @Override
            public void onError(String errorMessage, int statusCode) {
                Intent intent = new Intent(getBaseContext(), RecommendedRouteListActivity.class);
                startActivity(intent);
            }
        });

    }

    private TATRegion stringToRegion(String selectedRegion) {
        TATRegion region = null;
        switch (selectedRegion) {
            case "Northern":
                region = TATRegion.NORTHERN;
                return region;
            case "Northeastern":
                region = TATRegion.NORTHEASTERN;
                return region;
            case "Western":
                region = TATRegion.WESTERN;
                return region;
            case "Central":
                region = TATRegion.CENTRAL;
                return region;
            case "Eastern":
                region = TATRegion.EASTERN;
                return region;
            case "Southern":
                region = TATRegion.SOUTHERN;
                return region;
            default:
                return null;
        }
    }

}

File: app/src/main/java/org/tat/sdksample/recommendedroute/search/RecommendedRouteActivity.kt
package org.tat.sdksample.recommendedroute.search

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.MenuItem
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.Spinner
import android.widget.Switch
import org.tat.sdksample.R
import org.tat.sdksample.recommendedroute.list.RecommendedRouteListActivity
import org.tat.sdksample.recommendedroute.model.RecRoutes
import org.th.tatsdk.ServiceRequestListener
import org.th.tatsdk.common.TATGeolocation
import org.th.tatsdk.common.TATLanguage
import org.th.tatsdk.common.TATRegion
import org.th.tatsdk.route.TATFindRoutesParameter
import org.th.tatsdk.route.TATFindRoutesResultSet
import org.th.tatsdk.route.TATRecommendedRoutes

class RecommendedRouteActivity : AppCompatActivity() {

    private lateinit var spinnerDay: Spinner
    private lateinit var spinnerRegion: Spinner
    private lateinit var swNearby: Switch
    private lateinit var btnSearch: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_recommended_route)
        initInstances()
        setupToolbar()
    }

    private fun setupToolbar() {
        val actionBar = supportActionBar
        if (actionBar != null) {
            actionBar.setTitle(R.string.recommended_routes)
            actionBar.setHomeButtonEnabled(true)
            actionBar.setDisplayHomeAsUpEnabled(true)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> finish()
            else -> {
            }
        }
        return true
    }

    private fun initInstances() {
        spinnerDay = findViewById(R.id.spinnerDay)
        spinnerRegion = findViewById(R.id.spinnerRegion)
        swNearby = findViewById(R.id.swNearbyRecRoute)
        btnSearch = findViewById(R.id.btn_search_rectoute)
        btnSearch.setOnClickListener { loadRecRoute() }
        val days = resources.getStringArray(R.array.day)
        spinnerDay.adapter = ArrayAdapter(baseContext, android.R.layout.simple_list_item_1, days)
        val region = resources.getStringArray(R.array.region)
        spinnerRegion.adapter = ArrayAdapter(baseContext, android.R.layout.simple_list_item_1, region)

    }

    private fun loadRecRoute() {
        val param = TATFindRoutesParameter(TATLanguage.ENGLISH)
        val region = spinnerRegion.selectedItem.toString()
        val day = spinnerDay.selectedItemPosition
        if (region != "All") {
            val selectedRegion = stringToRegion(region)
            if (selectedRegion != null) {
                param.region = selectedRegion
            }
        }

        //set number of trip day
        if (day != 0) {
            param.numberOfDays = day
        }

        //set search center location
        if (swNearby.isChecked) {
            param.geolocation = TATGeolocation(13.74918,100.55785)
        }

        TATRecommendedRoutes.findAsync(param, object : ServiceRequestListener<TATFindRoutesResultSet> {

            override fun onResponse(result: TATFindRoutesResultSet?) {
                result?.results?.let {
                    val res = RecRoutes().transform(it)
                    val intent = Intent(baseContext, RecommendedRouteListActivity::class.java)
                    intent.putExtra("route_list", res)
                    startActivity(intent)
                }?:let {
                    val intent = Intent(baseContext, RecommendedRouteListActivity::class.java)
                    startActivity(intent)
                }

            }

            override fun onError(errorMessage: String?, statusCode: Int) {
                val intent = Intent(baseContext, RecommendedRouteListActivity::class.java)
                startActivity(intent)
            }
        })

    }

    private fun stringToRegion(selectedRegion: String): TATRegion? {
        var region: TATRegion? = null
        when (selectedRegion) {
            "Northern" -> {
                region = TATRegion.NORTHERN
                return region
            }
            "Northeastern" -> {
                region = TATRegion.NORTHEASTERN
                return region
            }
            "Western" -> {
                region = TATRegion.WESTERN
                return region
            }
            "Central" -> {
                region = TATRegion.CENTRAL
                return region
            }
            "Eastern" -> {
                region = TATRegion.EASTERN
                return region
            }
            "Southern" -> {
                region = TATRegion.SOUTHERN
                return region
            }
            else -> return null
        }
    }

}
